system/HTTP/CURLRequest.php

```diff
@@ -518,18 +518,18 @@ protected function setResponseHeaders(array $headers = [])
     protected function setCURLOptions(array $curlOptions = [], array $config = [])
     {
         // Auth Headers
-        if (! empty($config['auth'])) {
+        if (array_key_exists('auth', $config) && $config['auth']) {
             $curlOptions[CURLOPT_USERPWD] = $config['auth'][0] . ':' . $config['auth'][1];
 
-            if (! empty($config['auth'][2]) && strtolower($config['auth'][2]) === 'digest') {
+            if (array_key_exists(2, $config['auth']) && $config['auth'][2] && strtolower($config['auth'][2]) === 'digest') {
                 $curlOptions[CURLOPT_HTTPAUTH] = CURLAUTH_DIGEST;
             } else {
                 $curlOptions[CURLOPT_HTTPAUTH] = CURLAUTH_BASIC;
             }
         }
 
         // Certificate
-        if (! empty($config['cert'])) {
+        if (array_key_exists('cert', $config) && $config['cert']) {
```

Comment:
for right side of && we can check against empty array or empty string?

Reply:
:532   Result of || is always true.
         🪪  booleanOr.alwaysTrue
  :532   Strict comparison using !== between array{} and '' will always evaluate to true.
         🪪  notIdentical.alwaysTrue